src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -428,74 +526,170 @@ public static RequestDelegate Create(MethodInfo methodInfo, Func<HttpContext, ob
-                var lambda = Expression.Lambda<Func<object?, HttpContext, Task>>(body, TargetArg, HttpContextParameter);
-                var invoker = lambda.Compile();
-
-                requestDelegate = invoker;
+                return Expression.Lambda<Func<object?, HttpContext, Task>>(
+                    responseWritingMethodCall, TargetExpr, HttpContextExpr).Compile();
             }
-
-            return requestDelegate;
         }
 
-        private static ILogger GetLogger(HttpContext httpContext)
+        private static MethodInfo GetEnumTryParseMethod()
         {
-            var loggerFactory = httpContext.RequestServices.GetRequiredService<ILoggerFactory>();
-            return loggerFactory.CreateLogger("Microsoft.AspNetCore.Routing.MapAction");
-        }
-
-        private static Expression BindParamenter(Expression sourceExpression, ParameterInfo parameter, string? name)
-        {
-            var key = name ?? parameter.Name;
-            var type = Nullable.GetUnderlyingType(parameter.ParameterType) ?? parameter.ParameterType;
-            var valueArg = Expression.Convert(
-                                Expression.MakeIndex(sourceExpression,
-                                                     sourceExpression.Type.GetProperty("Item"),
-                                                     new[] { Expression.Constant(key) }),
-                                typeof(string));
+            var staticEnumMethods = typeof(Enum).GetMethods(BindingFlags.Public | BindingFlags.Static);
 
-            MethodInfo parseMethod = (from m in type.GetMethods(BindingFlags.Public | BindingFlags.Static)
-                                      let parameters = m.GetParameters()
-                                      where m.Name == "Parse" && parameters.Length == 1 && parameters[0].ParameterType == typeof(string)
-                                      select m).FirstOrDefault()!;
+            foreach (var method in staticEnumMethods)
+            {
+                if (!method.IsGenericMethod || method.Name != "TryParse" || method.ReturnType != typeof(bool))
+                {
+                    continue;
+                }
 
-            Expression? expr = null;
+                var tryParseParameters = method.GetParameters();
 
-            if (parseMethod != null)
-            {
-                expr = Expression.Call(parseMethod, valueArg);
+                if (tryParseParameters.Length == 2 &&
+                    tryParseParameters[0].ParameterType == typeof(string) &&
+                    tryParseParameters[1].IsOut)
+                {
+                    return method;
+                }
             }
-            else if (parameter.ParameterType != valueArg.Type)
+
+            throw new Exception("static bool System.Enum.TryParse<TEnum>(string? value, out TEnum result) does not exist!!?!?");
+        }
+
+        // TODO: Use InvariantCulture where possible? Or is CurrentCulture fine because it's more flexible?
```

Comment:
I'm interested in feedback on this. One wrinkle is there isn't a standard convention for `TryParse` methods taking `IFormatProvider`. For example, `int.TryParse` and similar want a `NumberStyles` argument as well for any overload taking `IFormatProvider`. We can special-case built-in numeric types of course.

Reply:
This is a topic that's up for debate. I think MVC does different things for different parts of the request.